### PR TITLE
These MultiInterpolation functions GriddedDataReporter

### LINKED ISSUE #47: function that uses griddedDataReporter
modify PiecewiseMultiInterpolationFromReporter to use gridded data output from the GriddedDataReporter and not from the GriddedData class.  Make PiecewiseMultilinearFromReporter evaluate a values for the function from a reporter and not from GriddedData.  This will allow us to change the griddedData through transfers.

### PR DESCRIPTION
**Pull request by lynnmunday**
closes #47

### CODE REVIEW

**Comment by lynnmunday** [comment]
@aaelmeli will you look at this.  @zachmprince Will you review this too

**Comment by lynnmunday** [comment]
@aaelmeli would you review this?

**Review by aaelmeli — CHANGES_REQUESTED** [review]
Can we add some example where the parameter values is supplied by some reporter that is different that the reporter that supplies the grids, etc.

File: src/functions/PiecewiseMultiInterpolationFromReporter.C [hunk 1]
```diff
@@ -8,29 +8,34 @@
...
-      "reporter z-coordinate name.  This uses the reporter syntax <reporter>/<name>.");
-  params.addRequiredParam<ReporterName>(
-      "time_name", "reporter time name.  This uses the reporter syntax <reporter>/<name>.");
-  params.addRequiredParam<ReporterName>(
-      "value_name", "reporter value name.  This uses the reporter syntax <reporter>/<name>.");
+      "dim_name",
+      "reporter from GriddedDataReporter containing grid dimension.  This "
+      "uses the reporter syntax <reporter>/<name>.");
+  params.addRequiredParam<ReporterName>("values_name",
+                                        "reporter containing value for each point in the grid.  "
```
[the remark this answers is not in the document]
Comment (lynnmunday) [reply src/functions/PiecewiseMultiInterpolationFromReporter.C:37]: I wanted it to be clear that the function values can from any reporter and not just those read in by the griddedDataReporter

**Pull request merged**